wakuv2/message_publishing.go

```diff
@@ -125,6 +123,9 @@ func (w *Waku) broadcast() {
 			}
 		}
 
+		// Wraps the publish function with rate limiter
+		fn = w.limiter.ThrottlePublishFn(w.ctx, fn)
```

Comment:
@richard-ramos I moved the rate limit wrapper after the telemetry wrapper so that rate limiting doesn't get caught as a publish error